sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteTables.java

```diff
@@ -176,13 +176,23 @@ public void processElement(ProcessContext c, BoundedWindow window) throws Except
               c.sideInput(loadJobIdPrefixView), tableDestination, partition, c.pane().getIndex());
 
       if (!singlePartition) {
+        // This is a temp table. Create a new one for each partition and each pane.
         tableReference.setTableId(jobIdPrefix);
       }
 
-      WriteDisposition writeDisposition =
-          (c.pane().getIndex() == 0) ? firstPaneWriteDisposition : WriteDisposition.WRITE_APPEND;
-      CreateDisposition createDisposition =
-          (c.pane().getIndex() == 0) ? firstPaneCreateDisposition : CreateDisposition.CREATE_NEVER;
+      WriteDisposition writeDisposition = firstPaneWriteDisposition;
+      CreateDisposition createDisposition = firstPaneCreateDisposition;
+      if (c.pane().getIndex() > 0 && singlePartition) {
+        // If writing directly to the destination, then the table is created on the first write
+        // and we should change the disposition for subsequent writes.
+        writeDisposition = WriteDisposition.WRITE_APPEND;
+        createDisposition = CreateDisposition.CREATE_NEVER;
+      } else if (!singlePartition) {
+        // In this case, we are writing to a temp table and always need to create it.
```

Comment:
Why does the comment say something different from the conditional? Can the structure be made more apparent? Can the comment explain why `!singlePartition` implies temp table? (I imagine if I knew the whole connector I'd know the answer, but that's sort of my point)

Reply:
I think this variable is poorly named - I'll rename it to noTempTables. It's true that a single partition implies there are no temp tables (because we can then load directly to the user's final table), but it's confusing in the code.

Another point of confusion: these "partitions" are not BigQuery partitions, but rather file partitions that the connector creates when there are too many files for a single load job. To be fair, I think this code was written before BigQuery partitions existed.

Reply:
Renamed variable, and switched the sense.